docs/guides/building-forms.md

```diff
@@ -0,0 +1,387 @@
+- `<input>`: The form element that is widely used to structure the form fields. It depends significantly on the value of the `type` attribute. Input types can be `text`, `checkbox`, `email`, `radio`, and more.
+- `<button>`: It represents a clickable button that's used to submit the form data.
+
+### Form Validation
+
+A process that checks if the information provided by a user is correct or not. Form validation also ensures that the provided info is in the correct format (e.g. there's an @ in the email field). These are of two types:
+
+- **Client-side**: Validation is done in the browser
+- **Server-side**: Validation is done on the server
+
+Though both these types are equally important, for this guide, we'll focus on client-side validation only.
+
+Client-side validation is further categorized as:
+
+- **Built-in**: Uses HTML-based attributes like `required`, `type`, `minLength`, `maxLength`, `pattern` etc.
+- **JavaScript-based**: Validation that's coded with JavaScript.
+
+### Built-in Form Validation Using `required`, `type`, `minLength`, `maxLength`
+
+- `required`: Specifies which fields must be filled before submitting the form.
+- `type`: Tells whether the data should be a number, email id, text string, etc.
+- `minLength`: Decides the minimum length for the text data string.
+- `maxLength`: Decides the maximum length for the text data string.
+
+Let's modify the form w.r.t. to these attributes:
+
+```html
+<!-- HTML Form with Built-in Validation -->
+<form action="/send-data-here" method="post">
+  <label for="roll">Roll Number</label>
+  <input
+    type="text"
+    id="roll"
+    name="roll"
+    required
+    minlength="10"
+    maxlength="20"
+  />
+  <label for="name">Name:</label>
+  <input type="text" id="name" name="name" required />
+  <button type="submit">Submit</button>
+</form>
+```
+
+With these validation checks in place, when a user tries to submit an empty field for Last Name, it gives an error that pops right in the form field. Similarly, a roll number can only be entered if it's 10-20 characters long.
+
+![form validation](https://assets.vercel.com/image/upload/c_scale,w_675/v1643009088/nextjs/guides/building-forms/form-validation.jpg)
+
+### JavaScript-based Form Validation
+
+Form Validation is important to ensure that a user has submitted appropriate data. JavaScript offers an additional level of validation along with HTML native form attributes on the client side. Developers generally prefer validating form data through JavaScript because its data processing is faster when compared to server-side validation.
+
+Let's look at a simple example:
+
+```html
+<form onsubmit="return validateFormWithJS()">
+  <label for="name">Name:</label>
+  <input type="text" name="name" id="name" />
+
+  <label for="rollNumber">Roll Number:</label>
+  <input type="text" name="rollNumber" id="rollNumber" />
+
+  <button type="submit">Submit</button>
+</form>
```

Comment:
This example has the same fields as the above one, but they are in different places. Can we keep them consistent? Its easier to read and scan if we are incrementally adding content to the code blocks, rather than swapping the field values

Reply:
Done ☑️